jpype/_jthread.py

```diff
@@ -0,0 +1,68 @@
+import _jpype
+import jpype
+import jpype.imports
+from . import _jcustomizer
+
+
+@_jcustomizer.JImplementationFor('java.lang.Thread')
+class _JThread(object):
+    """ Customizer for ``java.land.Thread``
+
+    This adds addition JPype methods to java.lang.Thread to support
+    Python.
+    """
+
+    @classmethod
+    def isAttached(self):
+        """ Checks if a thread is attached to the JVM.
+
+        Python automatically attaches as daemon threads when a Java method is
+        called.  This creates a resource in Java for the Python thread. This
+        method can be used to check if a Python thread is currently attached so
+        that it can be disconnected prior to thread termination to prevent
+        leaks.
+
+        Returns:
+          True if the thread is attached to the JVM, False if the thread is
+          not attached or the JVM is not running.
+        """
+        return _jpype.isThreadAttachedToJVM()
+
+    @classmethod
```

Comment:
This should be staticmethod, if you do not need the class to be passed. But this is only a minor detail.

Reply:
I am not sure if the customizer takes staticmethod customizations properly.   I will have to verify.   I remember that when I was doing the ``__javainit__`` methods I ran into a wierd corner case that transferring certain annotated methods caused odd behaviors, but I don't recall which it was.